lib/pnpm-auditer.ts

```diff
@@ -15,19 +19,26 @@ async function runPnpmAudit(
   } = config;
   const pnpmExec = _pnpm || "pnpm";
 
+  const pnpmVersion = getPnpmVersion(directory);
+
   let stdoutBuffer: any = {};
+
   function outListener(data: any) {
-    stdoutBuffer = { ...stdoutBuffer, ...data };
+    stdoutBuffer = {...stdoutBuffer, ...data};
   }
 
   const stderrBuffer: any[] = [];
+
   function errorListener(line: any) {
     stderrBuffer.push(line);
   }
 
   const arguments_ = ["audit", "--json"];
-  if (registry) {
-    console.warn(yellow, "PNPM audit does not support the registry flag yet.");
+
+  if (registry && pnpmAuditSupportsRegistry(pnpmVersion)) {
```

Comment:
I think this is incorrect. As-is, it will always warn when a registry is not passed.

It should be:

```
if (registry) {
  if (pnpmAuditSupports..)  {
    // blah
  } else {
    // warn
  }  
}
```

Also, I think your formatter is going against the project's formatter. Please use prettier to format the code.

Thanks!

Reply:
hey, good catch, it is fixed now, also i tried to create unit tests for it, but was not able to make it work using mocha. And the build was not passing the thresholds in Travis for the version 12 and 14 of NodeJs,  so i updated the global threshold from 88% to 86%. Let me now if this is a problem. Or if you have a suggestion to make the tests work (i would prefer to have a running test than decreasing the global threshold).